File: Python/Arrays/L_minTimeDiff.py
def findMinDifference(timePoints):
    # Convert times to minutes
    minutes = []
    for time in timePoints:
        h, m = map(int, time.split(':'))
        minutes.append(h * 60 + m)
    
    # Sort times in ascending order
    minutes.sort()
    
    # Find minimum difference across adjacent elements
    min_diff = float('inf')
    for i in range(len(minutes) - 1):
        min_diff = min(min_diff, minutes[i + 1] - minutes[i])
    
    # Consider the circular difference between last and first element
    min_diff = min(min_diff, 24 * 60 - minutes[-1] + minutes[0])
    
    return min_diff
        

File: JavaScript/Array/L_minTimeDiff.js
/**
 * @param {string[]} timePoints
 * @return {number}
 */
var findMinDifference = function (timePoints) {
    // Convert times to minutes
    let minutes = timePoints.map((time) => {
        let [h, m] = time.split(":").map(Number);
        return h * 60 + m;
    });

    // Sort times in ascending order
    minutes.sort((a, b) => a - b);

    // Find minimum difference across adjacent elements
    let minDiff = Infinity;
    for (let i = 0; i < minutes.length - 1; i++) {
        minDiff = Math.min(minDiff, minutes[i + 1] - minutes[i]);
    }

    // Consider the circular difference between last and first element
    minDiff = Math.min(minDiff, 24 * 60 - minutes[minutes.length - 1] + minutes[0]);

    return minDiff;
};
